src/psyclone/tests/transformations_test.py

```diff
@@ -221,3 +219,126 @@ def test_regiontrans_wrong_children():
+    then it creates a profile node with no name information.
+
+    '''
+    class Dummy(object):
+        '''Dummy class that checks the name argument is None.
+
+        :param NonType parent: dummy to conform to ProfileNode arguments.
+        :param NonType children: dummy to conform to ProfileNode arguments.
+        :param NoneType name: the name argument that we are testing.
+
+        '''
+        def __init__(self, parent=None, children=None, name=None):
+            self._parent = parent
+            self._children = children
+            if name is None:
+                raise Exception("test passed")
+    monkeypatch.setattr("psyclone.profiler.ProfileNode", Dummy)
+    # Create and apply the transformation.
+    profile_trans = ProfileRegionTrans()
+    with pytest.raises(Exception) as excinfo:
+        _ = profile_trans.apply(dummy_nodes())
+    assert "test passed" in str(excinfo.value)
+
+
+def test_profilerregiontrans_otheroption(monkeypatch):
+    '''Check that an option not associated with the profile transformation
+    is ignored.
+
+    '''
+    class Dummy(object):
+        '''Dummy class that checks the name argument is None.
+
+        :param NonType parent: dummy to conform to ProfileNode arguments.
+        :param NonType children: dummy to conform to ProfileNode arguments.
+        :param NoneType name: the name argument that we are testing.
+
+        '''
+        def __init__(self, parent=None, children=None, name=None):
+            self._parent = parent
+            self._children = children
+            if name is None:
+                raise Exception("test passed")
+    monkeypatch.setattr("psyclone.profiler.ProfileNode", Dummy)
+    # Create and apply the transformation.
+    profile_trans = ProfileRegionTrans()
+    options = {"someotheroption": "someothervalue"}
+    with pytest.raises(Exception) as excinfo:
+        _ = profile_trans.apply(dummy_nodes(), options=options)
+    assert "test passed" in str(excinfo.value)
+
+
+def test_profilerregiontrans_name(monkeypatch):
+    '''Check that when a valid name is supplied to the profile
+    transformation then it creates a profile node with that name
+    information.
+
+    '''
+    class Dummy(object):
+        '''Dummy class that checks the options argument is None.
+
+        :param NonType parent: dummy to conform to ProfileNode arguments.
+        :param NonType children: dummy to conform to ProfileNode arguments.
+        :param (str, str) name: the name argument we are testing.
+
+        '''
+        def __init__(self, parent=None, children=None, name=None):
+            self._parent = parent
+            self._children = children
+            assert name == ("x", "y")
+    monkeypatch.setattr("psyclone.profiler.ProfileNode", Dummy)
+    # Create and apply the transformation.
+    profile_trans = ProfileRegionTrans()
+    _ = profile_trans.apply(dummy_nodes(),
+                            options={"profile_name": ("x", "y")})
```

Comment:
Again, you could just interrogate the properties of the ProfileNode rather than having to define a dummy class?

Reply:
Yes, updated.